libsolidity/ast/Types.cpp

```diff
@@ -3494,7 +3497,7 @@ string FunctionType::externalSignature() const
 	}
 
 	// "inLibrary" is only relevant if this is not an event.
-	bool const inLibrary = kind() != Kind::Event && dynamic_cast<ContractDefinition const&>(*m_declaration->scope()).isLibrary();
+	bool const inLibrary = kind() != Kind::Event && m_declaration->scope() && dynamic_cast<ContractDefinition const&>(*m_declaration->scope()).isLibrary();
```

Comment:
I'm wondering whether we should rather assign scopes during parsing - it should be a purely syntactic thing, so the parser should be able to do that easily.

Reply:
Generally or for this PR?

Reply:
Generally.

Reply:
Just because I'm randomly reading these comments: I actually thought the same when looking at https://github.com/ethereum/solidity/pull/8834 - the ``m_free`` member there could just be derived from the scope, if the parser did this stuff already...